spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/autoconfig/GcpProperties.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core.autoconfig;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author João André Martins
+ */
+@ConfigurationProperties("cloud.gcp")
+public class GcpProperties {
+
+	private String projectId;
```

Comment:
I see a few reasons, but none of them very strong. 1) We might be interested in adding things to a project in the future, and then project.property would be recommended, which would be a configuration breaking change, 2) project.id looks better than projectId, 3) there are indeed other things in a project, some of them could be useful for other "legacy" APIs like the project number (but none that's critical besides ID). To have an idea of what a GCP project can have, you can refer to [this page](https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects). There are things like the IAM policy that we might want to use from Spring and that it's its own separate `Policy` entity in the GCP world, but would belong to a `Project` in Spring configuration.